### ISSUE TITLE
Typing in the end-date using date range mode affects the start-date

### ISSUE THREAD

**Issue by gdemartini** [eternicode/bootstrap-datepicker#1441]
Steps to reproduce:
- Set up 2 inputs to be used as date range selection:
![image](https://cloud.githubusercontent.com/assets/11699436/8240243/8ee82cd8-15d7-11e5-85c3-af5e0ab64881.png)
- Type in (using only the keyboard) the start date as 06/01/2015
![image](https://cloud.githubusercontent.com/assets/11699436/8240283/fb1bd620-15d7-11e5-840b-cc27bcd78f44.png)
- Tab to the next field, type in the end date as 06/18/2015. The moment you type the "2", it parses the text as a valid date which is smaller than the one currently in the start date field and changes the start date to "06/18/2".
![image](https://cloud.githubusercontent.com/assets/11699436/8240319/371ae094-15d8-11e5-91cf-19e5d38c18d1.png)
- The next keystrokes (015) have no effect on the start date, obviously.
![image](https://cloud.githubusercontent.com/assets/11699436/8240369/84935c0c-15d8-11e5-8555-0115b521036c.png)

**Comment by jangrewe** [comment]
+1

**Comment by divibyzero** [comment]
+1

I think it would be best if automatic change of the "start date" occurs only after user finishes editing the "end field" (that is - leaves focus of the input field). If no change is needed i.e. "end date" was typed in correctly - "start date" will remain unchanged.

**Comment by gioloi** [comment]
+1

**Comment by gioloi** [comment]
Ok... I found a workaround, I have changed the code to making sure that "changeDate" is triggered only when the length of the string in the input field is equal to the string of the date format set for datepicker (and not each time a digit is entered)... I'm using 1.4.0 version, I have changed (lines 724-726):

    ...
    else if (dates.length){
    // setting date by typing
    if (String(oldDates) !== String(this.dates))
        this._trigger('changeDate');
    ...

to

    ...
    else if (dates.length){
    // setting date by typing
    if ((String(this.element[0].value).length == String(this.o.format).length) && (String(oldDates) !== String(this.dates)))
       this._trigger('changeDate');
    ...

**Comment by steffendietz** [comment]
This seems to be a duplicate of #721

**Comment by hebbet** [comment]
thanks, closing that one.
